packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -145,6 +170,131 @@ export class RichTextEditor extends FoundationElement {
         return true;
     }
 
+    /**
+     * This function load tip tap editor with provided markdown content by parsing into html
+     * @public
+     */
+    public setMarkdown(value: string): void {
+        if (this.$fastController.isConnected) {
+            const htmlConvertedContent = this.getHtmlContent(value);
+            this.tiptapEditor.commands.setContent(htmlConvertedContent);
+            this.tiptapEditor.commands.focus();
+        }
+    }
+
+    /**
+     * This function returns markdown string by serializing tiptap editor document using prosemirror MarkdownSerializer
+     * @public
+     */
+    public getMarkdown(): string {
+        if (this.$fastController.isConnected) {
+            const markdownContent = this.markdownSerializer.serialize(
+                this.tiptapEditor.state.doc
+            );
+            return markdownContent;
+        }
+        return '';
+    }
+
+    /**
+     * This function takes the Fragment from parseMarkdownToDOM function and return the serialized string using XMLSerializer
+     */
+    private getHtmlContent(value: string): string {
+        const markdownDocumentFragment = this.parseMarkdownToDOM(value);
+        return this.xmlSerializer.serializeToString(markdownDocumentFragment);
+    }
+
+    private initializeMarkdownParser(): MarkdownParser {
```

Comment:
Moving the common parsing functionalities to a separate folder is in the pipeline, Will create subsequent PR for this implementation.

Reply:
I have a slight preference for having a folder structure of:

```
/rich-text
    /editor
        ... unshared models and existing content here
    /viewer
    /models
        ... shared models here
```

to avoid having a top-level empty `rich-text` folder assuming that doesn't have any other complication.